depthai_ros_driver/src/depthai_common.cpp

```diff
@@ -0,0 +1,245 @@
+#include <depthai_ros_driver/depthai_common.hpp>
+
+namespace rr {
+
+//==============================================================================
+const std::pair<std::string, cv_bridge::CvImage> get_image_data(
+    const HostDataPacket& packet,
+    const Stream& type)
+{
+    std::pair<std::string, cv_bridge::CvImage> img_data;
+    cv_bridge::CvImage cvImg;
+    std::string encoding = "";
+
+    const int rows = packet.dimensions[0];
+    const int cols = packet.dimensions[1];
+
+    // cv::Mat needs `void *` not `const void *`
+    auto* data = const_cast<uint8_t*>(packet.getData());
+
+    switch (type) {
+        case Stream::LEFT:
+        case Stream::RIGHT:
+        case Stream::RECTIFIED_LEFT:
+        case Stream::RECTIFIED_RIGHT:
+        case Stream::DISPARITY:
+            cvImg.image = cv::Mat(rows, cols, CV_8UC1, data);
+            encoding = "mono8";
+            break;
+        case Stream::PREVIEW_OUT: {
+            std::vector<cv::Mat> toMerge(3);
+            const auto offset = cols * cols;
+            for (int i = 0; i < 3; ++i) {
+                toMerge[i] = cv::Mat(cols, cols, CV_8UC1, data + i * offset);
+            }
+            cv::merge(toMerge, cvImg.image);
+            encoding = "bgr8";
+            break;
+        }
+        case Stream::JPEG_OUT:
+        case Stream::VIDEO: {
+            // const auto img = boost::make_shared<sensor_msgs::CompressedImage>();
```

Comment:
since this condition uses a different msg compare to other "Stream" (`CompressedImageMsg` instead of `ImgMsg`), i have moved the impl to the ros1/ros2 node class: [here](https://github.com/osrf/depthai_ros/blob/81f5615f1a7cbbafea0f0e702f95b11a7e668b8f/depthai_ros_driver/src/depthai_base_ros2.cpp#L175-L179)